packages/shared/src/hooks/useTagAndSource.ts

```diff
@@ -0,0 +1,108 @@
+import { useContext } from 'react';
+import AuthContext from '../contexts/AuthContext';
+import AnalyticsContext from '../contexts/AnalyticsContext';
+import useMutateFilters from './useMutateFilters';
+import { Source } from '../graphql/sources';
+
+interface TagActionArguments {
+  tags: Array<string>;
+  category?: string;
+}
+
+interface SourceActionArguments {
+  source: Source;
+}
+
+export default function useTagAndSource({ origin }: { origin?: string }): {
```

Comment:
If we block source or tag through a post menu context we should add the relevant post id in the extra property